llvm/include/llvm/Analysis/IR2Vec.h

```diff
@@ -53,7 +53,24 @@ class raw_ostream;
 enum class IR2VecKind { Symbolic };
 
 namespace ir2vec {
-using Embedding = std::vector<double>;
+/// Embedding is a ADT that wraps std::vector<double>. It provides
+/// additional functionality for arithmetic and comparison operations.
+struct Embedding : public std::vector<double> {
```

Comment:
I'm not sure inheritance is ideal here. How about containment, and implicit constructor from a std::vector?

Reply:
Sure. Right now we would need iterators and common APIs like size().